Server/src/Server.Application/CommandHandlers/AddBookToLibraryHandler.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Text.Json.Serialization;
+using FluentValidation;
+using MediatR;
+using Server.Application.Exceptions;
+using Server.Application.Exceptions.Types;
+using Server.Application.InfrastructureInterfaces;
+using Server.Domain.Entities;
+using Server.Domain.Repositories;
+
+namespace Server.Application.CommandHandlers;
+
+public sealed class AddBookToLibraryHandlerValidator : AbstractValidator<AddBookToLibraryCommand>
+{
+    public AddBookToLibraryHandlerValidator()
+    {
+        RuleFor(x => x.Amount).GreaterThanOrEqualTo(1);
+    }
+}
+
+public sealed record AddBookToLibraryCommand : IRequest
+{
+    public Guid BookId { get; set; }
+    public int Amount { get; set; }
+    [JsonIgnore]
+    public Guid LibraryId { get; set; }
+}
+
+public sealed class AddBookToLibraryHandler : IRequestHandler<AddBookToLibraryCommand>
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly ILibraryRepository _libraryRepository;
+    private readonly IBookRepository _bookRepository;
+    private readonly IBookInLibraryRepository _bookInLibraryRepository;
+
+    public AddBookToLibraryHandler(IUnitOfWork unitOfWork, ILibraryRepository libraryRepository,
+        IBookRepository bookRepository, IBookInLibraryRepository bookInLibraryRepository)
+    {
+        _unitOfWork = unitOfWork;
+        _libraryRepository = libraryRepository;
+        _bookRepository = bookRepository;
+        _bookInLibraryRepository = bookInLibraryRepository;
+    }
+
+    public async Task Handle(AddBookToLibraryCommand request, CancellationToken cancellationToken)
+    {
+        var library = await _libraryRepository.FirstOrDefaultByIdAsync(request.LibraryId, cancellationToken);
+        var book = await _bookRepository.FirstOrDefaultByIdAsync(request.BookId, cancellationToken);
+
+        if (library == null)
+        {
+            throw new NotFoundException($"Library with id: {request.LibraryId} not found", ApplicationErrorCodes.LibraryNotFound);
+        }
+
+        if (book == null)
+        {
+            throw new NotFoundException($"Book with id: {request.BookId} not found", ApplicationErrorCodes.BookNotFound);
+        }
+
+        var bookInLibrary = LibraryBook.Create(library, book, request.Amount);
```

Comment:
Przydałoby się chyba sprawdzać czy dana książka jest już w bibliotece i ewentualnie zwiększać samo jej Amount przy powtórzeniu lub rzucać błędem i oddzielna mutacja do dodawania ilości

Reply:
można sprawdzić, ale ogólnie na froncie można wybrać tylko te książki, które nie są jeszcze dodane do wybranej biblioteki (na razie wpisana ręcznie, potem będzie brana z claimów Usera). Więc nie wiem czy dodawać tu też czy nie?

Reply:
To już zależy, ogólnie powinniśmy to sprawdzać i nie polegać na froncie, ale nie jest to w sumie wymagane w tym projekcie skoro to nigdy nie będzie produkcyjne rozwiązanie xd.

Reply:
Dodałem i wydzieliłem jeszcze logikę pobierania danych z Handlerów, bo to chyba w repozytorium powinno być, możesz zerknąć